target_chains/solana/pyth_solana_receiver_state/src/price_update.rs

```diff
@@ -31,10 +52,87 @@ impl PriceUpdateV1 {
     pub const LEN: usize = 8 + 32 + 2 + 32 + 8 + 8 + 4 + 8 + 8 + 8 + 8;
 }
 
+pub struct Price {
+    pub price:        i64,
+    pub conf:         u64,
+    pub exponent:     i32,
+    pub publish_time: i64,
+}
+
+impl PriceUpdateV1 {
+    pub fn get_price_unchecked(
+        &self,
+        feed_id: &FeedId,
+    ) -> std::result::Result<Price, GetPriceError> {
+        check!(
+            self.price_message.feed_id == *feed_id,
+            GetPriceError::WrongFeedId
+        );
+        Ok(Price {
+            price:        self.price_message.price,
+            conf:         self.price_message.conf,
+            exponent:     self.price_message.exponent,
+            publish_time: self.price_message.publish_time,
+        })
+    }
+
+    pub fn get_price_no_older_than_with_custom_verification_level(
+        &self,
+        clock: &Clock,
+        maximum_age: u64,
+        feed_id: &FeedId,
+        verification_level: VerificationLevel,
+    ) -> std::result::Result<Price, GetPriceError> {
+        check!(
+            self.verification_level.gte(verification_level),
+            GetPriceError::WrongVerificationLevel
+        );
+        let price = self.get_price_unchecked(feed_id)?;
+        check!(
+            price
+                .publish_time
+                .saturating_add(maximum_age.try_into().unwrap())
+                >= clock.unix_timestamp,
+            GetPriceError::PriceTooOld
+        );
+        Ok(price)
+    }
+
+    pub fn get_price_no_older_than(
+        &self,
+        clock: &Clock,
+        maximum_age: u64,
+        feed_id: &FeedId,
+    ) -> std::result::Result<Price, GetPriceError> {
+        self.get_price_no_older_than_with_custom_verification_level(
+            clock,
+            maximum_age,
+            feed_id,
+            VerificationLevel::Full,
+        )
+    }
+}
+
+/**
+ * This function takes a hex string and returns a FeedId
+ * The hex string should be 66 characters long, and should start with "0x"
```

Comment:
suggest supporting both the 0x prefixed version and the non-0x prefixed. This is a common cause of confusion in many places.